### PR TITLE
[hashicorp_vault] Fix fields.yml definition for dynamic mapping

### LINKED ISSUE #7811: [hashicorp_vault] Convert dynamic mappings to fields.yml
Prior to package-spec supporting a means to add dynamic mappings through `fields/*.yml` files, the hashicorp_vault package created those dynamic mappings by directly specifying them through `elasticsearch.index_template.mappings` in the manifest.

Under package-spec 3.0.0 those fields are being treated as invalid. I think it should be possible to express those fields in a `fields/*.yml` file now. So let's do that to resolve these errors.

```
Error: building package failed: invalid content found in built zip package: found 4 validation errors:
   1. file "hashicorp_vault-1.14.0.zip/data_stream/metrics/manifest.yml" is invalid: field elasticsearch.index_template.mappings.dynamic_templates.0.all_values: Additional property path_match is not allowed
   2. file "hashicorp_vault-1.14.0.zip/data_stream/metrics/manifest.yml" is invalid: field elasticsearch.index_template.mappings.dynamic_templates.1.all_counters: Additional property path_match is not allowed
   3. file "hashicorp_vault-1.14.0.zip/data_stream/metrics/manifest.yml" is invalid: field elasticsearch.index_template.mappings.dynamic_templates.2.all_rates: Additional property path_match is not allowed
   4. file "hashicorp_vault-1.14.0.zip/data_stream/metrics/manifest.yml" is invalid: field elasticsearch.index_template.mappings.dynamic_templates.3.all_histograms: Additional property path_match is not allowed
```

Testing should be performed to compare the installed Elasticsearch index template before and after the change.

### PR DESCRIPTION
**Pull request by andrewkroh**
## What does this PR do?

Earlier versions of Fleet did not handle dynamic mappings correctly so hashicorp_vault explicitly specified the mappings through the manifest. This is no longer needed for 8.x. Fixes #7811

Also refer to the ECS definition for all fields that exist in ECS.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Relates  #7811
- Depends on: #7960  (to make pipeline tests pass, otherwise unrelated to these changes)

## Mappings

This is how the mappings changed.

```diff
2d1
<   "dynamic": true,
5c4
<       "all_values": {
---
>       "hashicorp_vault.metrics.*.value": {
8a8
>         "match_mapping_type": "*",
13c13
<       "all_counters": {
---
>       "hashicorp_vault.metrics.*.counter": {
16a17
>         "match_mapping_type": "*",
21c22
<       "all_rates": {
---
>       "hashicorp_vault.metrics.*.rate": {
24a26
>         "match_mapping_type": "*",
29c31
<       "all_histograms": {
---
>       "hashicorp_vault.metrics.*.histogram": {
32a35
>         "match_mapping_type": "*",
79,94d81
<         }
<       }
<     },
<     "hashicorp_vault": {
<       "properties": {
<         "metrics": {
<           "properties": {
<             "*": {
<               "properties": {
<                 "*": {
<                   "dynamic": true,
<                   "type": "object"
<                 }
<               }
<             }
<           }
```

### CODE REVIEW

File: packages/hashicorp_vault/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.16.0-next"
```
Comment (andrewkroh) [inline packages/hashicorp_vault/changelog.yml:2]: I indend to open a few separate PRs to address other issues and bring this up to format_version 3.0.0.

File: packages/hashicorp_vault/data_stream/metrics/fields/fields.yml [hunk 2]
```diff
@@ -1,5 +1,17 @@
...
+  description: Hashicorp Vault telemetry data from the Prometheus endpoint.
+- name: hashicorp_vault.metrics.*.counter
+  type: double
+  object_type_mapping_type: '*'
+  description: Hashicorp Vault telemetry data from the Prometheus endpoint.
+- name: hashicorp_vault.metrics.*.rate
+  type: double
+  object_type_mapping_type: '*'
+  description: Hashicorp Vault telemetry data from the Prometheus endpoint.
+- name: hashicorp_vault.metrics.*.histogram
```
Comment (efd6) [inline packages/hashicorp_vault/data_stream/metrics/fields/fields.yml:13]: Should this not also have a `object_mapping_type` decl?

Comment (andrewkroh) [reply packages/hashicorp_vault/data_stream/metrics/fields/fields.yml:13]: It probably should too just to be explicit and prevent regression if Fleet decides to change its behavior. The Fleet generated mapping for histograph (and only histogram) had `match_mapping_type: '*'` which is what I want. That matches the behavior of what it had before where no `match_mapping_type` was given.

I will add it.

Comment (andrewkroh) [reply packages/hashicorp_vault/data_stream/metrics/fields/fields.yml:13]: Add in bb8160b.

**Review by efd6 — APPROVED** [review]

**Pull request merged**